pkg/process/util/network_filter.go

```diff
@@ -0,0 +1,73 @@
+package util
+
+import (
+	"net"
+	"strconv"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// ConnectionFilter holds a user-defined blacklisted IP/CIDR, and ports
+type ConnectionFilter struct {
+	IP    *net.IPNet
+	Ports map[uint16]struct{}
+}
+
+// ParseBlacklist takes the user defined blacklist and returns a slice of ConnectionFilters
+func ParseBlacklist(filters map[string][]string) (blacklist []*ConnectionFilter) {
+	for ip, ports := range filters {
+		conn := &ConnectionFilter{Ports: map[uint16]struct{}{}}
+
+		// retrieve valid IPs
+		if strings.Contains(ip, "/") {
```

Comment:
```
        var subnet *IPNet
        var err error

		if strings.Contains(ip, "/") {
			_, subnet, err = net.ParseCIDR(ip)
		} else if strings.Contains(ip, ":") {
			_, subnet, err = net.ParseCIDR(ip + "/64") // if given ipv6, prefix length of 64
		} else {
			_, subnet, err := net.ParseCIDR(ip + "/32") // if given ipv4, prefix length of 32
		}

		if err != nil || subnet == nil {
			log.Debugf("Could not parse given IPs: %s", err)
            continue
		}
		
        conn.IP = subnet
        ...
```

Reply:
We should make sure we short-circuit if the subnet could not be parsed.

Also we can clean this up and remove duplication.